storage/src/helpers/block_locators.rs

```diff
@@ -87,26 +86,17 @@ impl<N: Network> FromBytes for BlockLocators<N> {
             if header_exists {
                 let mut buffer = vec![0u8; N::HEADER_SIZE_IN_BYTES];
                 reader.read_exact(&mut buffer)?;
-                block_headers_bytes.push((height, buffer));
-            };
-
-            block_locators.insert(height, (hash, None));
-        }
-
-        let block_headers = block_headers_bytes
-            .par_iter()
-            .flat_map(|(height, bytes)| match !bytes.is_empty() {
-                true => Some((height, BlockHeader::<N>::read_le(&bytes[..]).unwrap())),
-                false => None,
-            })
-            .collect::<Vec<_>>();
-
-        for (height, block_header) in block_headers.into_iter() {
-            if let Some((_, header)) = block_locators.get_mut(height) {
-                *header = Some(block_header);
+                block_headers_bytes.push((height, hash, Some(buffer)));
+            } else {
+                block_headers_bytes.push((height, hash, None));
             }
         }
 
+        let block_locators = block_headers_bytes
+            .into_par_iter()
+            .map(|(height, hash, bytes)| (height, (hash, bytes.map(|bytes| BlockHeader::<N>::read_le(&bytes[..]).unwrap()))))
```

Comment:
Might be worth adding a note about why the `unwrap` is safe here?

Reply:
I only preserved the earlier behavior here; I don't know if it can be guaranteed.